### PR TITLE
Introduce `convert_to_coding_scheme` and make `coding_scheme` deprecated in CoNLL2003DatasetReader

### LINKED ISSUE #1941: IOB1 -> BIOUL conversion happens even when data is already in BIOUL
**Describe the bug**
When training the NER model on CoNLL files with BIOUL NER tags and specifying the the coding scheme as such, an error is thrown when to_bioul() is called on something that is already formatted in BIOUL

**To Reproduce**
Steps to reproduce the behavior
1. Include the following in the config:
```
"dataset_reader": {
    "type": "conll2003",
    "tag_label": "ner",
    "coding_scheme": "BIOUL",
     ....
  },

```
2. run ``allennlp train`` on the config with a BIOUL-labeled CoNLL input file
3. receive the following error:
```
0it [00:00, ?it/s]2018-10-23 15:59:50,380 - INFO - allennlp.data.dataset_readers.conll2003 - Reading instances from lines in file at: ../data/train/ner-conll/320_extraction.conll
Traceback (most recent call last):
  File "/anaconda3/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/anaconda3/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/anaconda3/lib/python3.7/site-packages/allennlp/run.py", line 18, in <module>
    main(prog="allennlp")
  File "/anaconda3/lib/python3.7/site-packages/allennlp/commands/__init__.py", line 70, in main
    args.func(args)
  File "/anaconda3/lib/python3.7/site-packages/allennlp/commands/train.py", line 102, in train_model_from_args
    args.recover)
  File "/anaconda3/lib/python3.7/site-packages/allennlp/commands/train.py", line 132, in train_model_from_file
    return train_model(params, serialization_dir, file_friendly_logging, recover)
  File "/anaconda3/lib/python3.7/site-packages/allennlp/commands/train.py", line 261, in train_model
    all_datasets = datasets_from_params(params)
  File "/anaconda3/lib/python3.7/site-packages/allennlp/commands/train.py", line 149, in datasets_from_params
    train_data = dataset_reader.read(train_data_path)
  File "/anaconda3/lib/python3.7/site-packages/allennlp/data/dataset_readers/dataset_reader.py", line 73, in read
    instances = [instance for instance in Tqdm.tqdm(instances)]
  File "/anaconda3/lib/python3.7/site-packages/allennlp/data/dataset_readers/dataset_reader.py", line 73, in <listcomp>
    instances = [instance for instance in Tqdm.tqdm(instances)]
  File "/anaconda3/lib/python3.7/site-packages/tqdm/_tqdm.py", line 937, in __iter__
    for obj in iterable:
  File "/anaconda3/lib/python3.7/site-packages/allennlp/data/dataset_readers/conll2003.py", line 119, in _read
    yield self.text_to_instance(tokens, pos_tags, chunk_tags, ner_tags)
  File "/anaconda3/lib/python3.7/site-packages/allennlp/data/dataset_readers/conll2003.py", line 139, in text_to_instance
    encoding=self._original_coding_scheme) if ner_tags is not None else None
  File "/anaconda3/lib/python3.7/site-packages/allennlp/data/dataset_readers/dataset_utils/span_utils.py", line 367, in to_bioul
    raise InvalidTagSequence(tag_sequence)
allennlp.data.dataset_readers.dataset_utils.span_utils.InvalidTagSequence: O O O O O B-DATA L-DATA O O O O O
```

**Expected behavior**
The trainer should run as expected.

**System (please complete the following information):**
 - OS: MacOS
 - Python version: 3.7
 - AllenNLP version: I installed from master

**Additional context**
Not defining the coding scheme in the config solves the issue, but it isn't clearly documented (as far as I could tell) that this would be the behavior of specifying BIOUL as the coding scheme. It's probably not super important as I'm told the generally accepted format in CoNLL files is IOB1 anyway but @joelgrus recommended I open an issue for it to at least start the conversation.

### PR DESCRIPTION
**Pull request by himkt**
Fixes #1941 and retry of #4924.

This PR introduces `convert_to_coding_scheme` to `Conll2003DatasetReader` based on the [suggestion](https://github.com/allenai/allennlp/pull/4924#discussion_r566309610) by @epwalsh.

### CODE REVIEW

File: allennlp/data/dataset_readers/conll2003.py [hunk 1]
```diff
@@ -66,13 +67,19 @@ class Conll2003DatasetReader(DatasetReader):
         Each will have its own namespace : `pos_tags`, `chunk_tags`, `ner_tags`.
         If you want to use one of the tags as a `feature` in your model, it should be
         specified here.
-    coding_scheme : `str`, optional (default=`IOB1`)
+    convert_to_coding_scheme : `str`, optional (default=`None`)
         Specifies the coding scheme for `ner_labels` and `chunk_labels`.
-        Valid options are `IOB1` and `BIOUL`.  The `IOB1` default maintains
+        `Conll2003DatasetReader` assumes a coding scheme of input data is `IOB1`.
+        Valid options are `None` and `BIOUL`.  The `None` default maintains
```
Comment (himkt) [inline allennlp/data/dataset_readers/conll2003.py:73]: https://github.com/allenai/allennlp/pull/4924#discussion_r566305479

@epwalsh 
I added a note about the assumed coding scheme.
Does this change match your intention?

**Comment by himkt** [comment]
@epwalsh Sorry for the delay. I opened the new PR because the original PR contains changes in the fixture.
I revised the PR based on your suggestions. Please take a look.

File: allennlp/data/dataset_readers/conll2003.py [hunk 2]
```diff
@@ -84,10 +91,24 @@ def __init__(
...
-        coding_scheme: str = "IOB1",
+        convert_to_coding_scheme: Optional[str] = None,
         label_namespace: str = "labels",
         **kwargs,
     ) -> None:
+
+        if "coding_scheme" in kwargs:
+            warnings.warn("`coding_scheme` is deprecated.", DeprecationWarning)
+            coding_scheme = kwargs["coding_scheme"]
+            kwargs.pop("coding_scheme")
```
Comment (epwalsh) [inline allennlp/data/dataset_readers/conll2003.py:102]:
```suggestion
            coding_scheme = kwargs.pop("coding_scheme")
```

File: tests/data/dataset_readers/conll2003_test.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        tokens = [t.text for t in fields["tokens"].tokens]
+        assert tokens == ["U.N.", "official", "Ekeus", "heads", "for", "Baghdad", "."]
+        assert fields["tags"].labels == expected_labels
+
+        fields = instances[1].fields
+        tokens = [t.text for t in fields["tokens"].tokens]
+        assert tokens == ["AI2", "engineer", "Joel", "lives", "in", "Seattle", "."]
+        assert fields["tags"].labels == expected_labels
+
+    @pytest.mark.filterwarnings("ignore::DeprecationWarning")
```
Comment (epwalsh) [inline tests/data/dataset_readers/conll2003_test.py:33]: I don't think we need this here, right?

```suggestion
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+- Introduce `convert_to_coding_scheme` and make `coding_scheme` deprecated in `Conll2003DatasetReader`.
```
Comment (epwalsh) [inline CHANGELOG.md:10]:
```suggestion
### Changed

- `coding_scheme` parameter is now deprecated in `Conll2003DatasetReader`, please use `convert_to_coding_scheme` instead.
```

**Review by epwalsh — COMMENTED** [review]
Looks great @himkt! Just a few minor suggestions

File: tests/data/dataset_readers/conll2003_test.py (hunk 3 above)
Comment (himkt) [reply tests/data/dataset_readers/conll2003_test.py]: You're right. Thank you.

**Review by epwalsh — APPROVED** [review]
LGTM!

**Pull request merged**